server/src/main/java/org/elasticsearch/cluster/coordination/LeaderChecker.java

```diff
@@ -234,16 +234,19 @@ public void handleException(TransportException exp) {
                         }
 
                         if (exp instanceof ConnectTransportException || exp.getCause() instanceof ConnectTransportException) {
-                            logger.debug(new ParameterizedMessage("leader [{}] disconnected, failing immediately", leader), exp);
-                            leaderFailed();
+                            logger.debug(new ParameterizedMessage(
+                                "leader [{}] disconnected during check, restarting discovery", leader), exp);
```

Comment:
"restarting discovery" does not belong to a message in this class (it assumes stuff about calling context). Perhaps just leave that part out

Reply:
Good point, this also moved to the message in `Coordinator`. Fixed in 
f55db2a.